test/widgets/profile/edit_brith_date_page_test.dart

```diff
@@ -0,0 +1,136 @@
+
+      expect(birthDateFinder, findsOneWidget);
+      expect(birthDateInputFinder, findsOneWidget);
+    });
+    testWidgets('change birthDate', (WidgetTester tester) async {
+      await pumpMaterial(tester, EditBirthDatePage(profile));
+      await tester.pump();
+
+      final Finder birthDateFinder = find.text(localeManager.formatDate(profile.birthDate!));
+      final Finder birthDateInputFinder = find.byKey(const Key('birthDateInput'));
+
+      await tester.tap(birthDateInputFinder);
+      await tester.pumpAndSettle();
+      expect(find.byType(DatePickerDialog), findsOneWidget);
+
+      await tester.tap(find.byIcon(Icons.edit));
+      await tester.pumpAndSettle();
+
+      await tester.tap(birthDateFinder.last);
+      final String date = localeManager.formatDate(profile.birthDate!.add(const Duration(days: 5)));
+
+      await tester.enterText(birthDateFinder.last, date);
+      await tester.tap(find.text('OK'));
+      await tester.pumpAndSettle();
+      expect(find.text(date), findsOneWidget);
+    });
+    testWidgets('clear Button', (WidgetTester tester) async {
+      await pumpMaterial(tester, EditBirthDatePage(profile));
+      await tester.pump();
+
+      final Finder birthDateFinder = find.text(localeManager.formatDate(profile.birthDate!));
+
+      expect(birthDateFinder, findsOneWidget);
+
+      final Finder clearButton = find.byKey(const Key('clearButton'));
+      expect(clearButton, findsOneWidget);
+
+      await tester.tap(clearButton);
+      await tester.pump();
+      expect(birthDateFinder, findsNothing);
+    });
+    testWidgets('save Button', (WidgetTester tester) async {
+      await pumpMaterial(tester, ProfilePage.fromProfile(profile));
+      await tester.pump();
+      expect(find.text(profile.age.toString()), findsOneWidget);
+
+      await tester
+          .tap(find.descendant(of: find.byKey(const Key('age')), matching: find.byKey(const Key('editButton'))));
+      await tester.pumpAndSettle();
+      expect(find.byType(EditBirthDatePage), findsOneWidget);
+
+      final Finder clearButton = find.byKey(const Key('clearButton'));
+      await tester.tap(clearButton);
+      await tester.pump();
+
+      final Finder saveButtonFinder = find.byKey(const Key('saveButton'));
+      expect(saveButtonFinder, findsOneWidget);
+      await tester.tap(saveButtonFinder);
+      await tester.pumpAndSettle();
+
+      expect(find.byType(ProfilePage), findsOneWidget);
+
+      verifyRequest(
+        processor,
+        urlMatcher: equals('/rest/v1/profiles?id=eq.1'),
+        methodMatcher: equals('PATCH'),
+        bodyMatcher: equals({'birth_date': null}),
+      ).called(1);
+
+      verifyRequest(processor,
+              urlMatcher: equals(
+                  '/rest/v1/profiles?select=%2A%2Cprofile_features%28%2A%29%2Creviews_received%3Areviews%21reviews_receiver_id_fkey%28%2A%2Cwriter%3Awriter_id%28%2A%29%29%2Creports_received%3Areports%21reports_offender_id_fkey%28%2A%29&id=eq.1'))
```

Comment:
okay you don't need to match the whole URL. `startsWith('/rest/v1/profiles')` is enough. Or maybe a regex to capture the id at the end, but that would probably be overkill.